### PR TITLE
Fixed create recipe

### PR DESCRIPTION
**Pull request by KiyaniBamba**
# Description

Please include a summary of the change and a link to which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.
Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Change status

- [ ] Complete, tested, ready to review and merge
- [ ] Complete, but not tested (may need new tests)
- [ ] Incomplete/work-in-progress, PR is for discussion/feedback

# How Has This Been Tested?

Please describe steps taken to ensure this feature is functional and does not break other features:

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation, if relevant
- [ ] I have run the app using my feature and ensured that no functionality is broken
- [ ] My changes generate no new warnings
- [ ] There are no merge conflicts

### CODE REVIEW

**Comment by KiyaniBamba** [comment]
The problem concerning the image, instructions and ingredients upload has been fixed. However, I have a new bug when I tried to do a preview of the recipe. 
Goforward between step 4 and 5 doesn't work when you create a recipe from the start, but it works when you just add instructions.

File: src/components/createRecipe/createRecipe/CreateRecipe.jsx [hunk 1]
```diff
@@ -10,12 +10,11 @@ import Footer from "../../navigation/footerNav/FooterNav";
 
 function App(props) {
 
-  const [page, setPage] = useState(3);
+  const [page, setPage] = useState(4);
```
Comment (carnundotcom) [inline src/components/createRecipe/createRecipe/CreateRecipe.jsx]: We'll want this to be 1 by default, no? (Or 0, or wherever it starts counting from?)

File: src/components/createRecipe/recipeForm/Step2.jsx [hunk 2]
```diff
@@ -13,30 +16,48 @@ import {
...
 function Step2(props) {
+
+    const [loading, setLoading] = useState(false);
+  const uploadImageToCloud = e => {
+    imageUpload(e, setLoading, setImgUrl);
+  };
   const [imgUrl, setImgUrl] = useState(false);
-  const [loading, setLoading] = useState(false);
-  const { goForward, goBackward, addImagesToBody } = props;
+  const { goForward,
```
Comment (carnundotcom) [inline src/components/createRecipe/recipeForm/Step2.jsx]: Should make a new line here. Formatting is a little funky!

File: src/components/createRecipe/recipeForm/Step2.jsx [hunk 3]
```diff
@@ -13,30 +16,48 @@ import {
...
-  const { goForward, goBackward, addImagesToBody } = props;
+  const { goForward, 
+        addImagesToBody, 
+        goBackward 
+      } = props;
 
   const onSubmit = e => {
     e.preventDefault();
-    addImagesToBody(imgUrl);
+    addImagesToBody(imgUrl)
```
Comment (carnundotcom) [inline src/components/createRecipe/recipeForm/Step2.jsx]: Why remove the semicolon?

File: src/components/createRecipe/recipeForm/Step2.jsx [hunk 4]
```diff
@@ -13,30 +16,48 @@ import {
...
+      };
+
+  const uploadImage = async e => {
+    e.preventDefault();
+    try {
+      const files = e.target.files;
+      const data = new FormData();
+      data.append("file", files[0]);
+      data.append("upload_preset", "recipe_image");
+      const imageUrl = await axios .post("https://api.cloudinary.com/v1_1/dr34bum3p/image/upload", data)
```
Comment (carnundotcom) [inline src/components/createRecipe/recipeForm/Step2.jsx]: There's a space here between `axios` and `.post()`?

File: src/components/createRecipe/recipeForm/Step2.jsx [hunk 5]
```diff
@@ -52,17 +73,18 @@ function Step2(props) {
...
         </NavigationSection1>
         <Addtitle>
           <h1>Upload Image</h1>
         </Addtitle>
       </Section3>
-
       <ExportImg>
         <div>
-          {imgUrl ? (
+           {imgUrl ? (
```
Comment (carnundotcom) [inline src/components/createRecipe/recipeForm/Step2.jsx:88]: ?

File: src/components/createRecipe/recipeForm/Step4.jsx [hunk 6]
```diff
@@ -17,16 +12,28 @@ import {
...
+
   const [instructionsArray, setInstructionsArray] = useState([]);
 
   const inputHandler = e => {
     e.preventDefault();
     setInputState(e.target.value);
   };
 
+
+  const goBack = e => {
```
Comment (carnundotcom) [inline src/components/createRecipe/recipeForm/Step4.jsx:33]: I've seen this function a few times now. Perhaps it should be passed down in props, not redefined within each component?

File: src/components/createRecipe/recipeForm/Step4.jsx [hunk 7]
```diff
@@ -96,30 +97,34 @@ function Step4(props) {
...
-          InputProps={{ classes: { root: classes.inputRoot2 } }}
-          className={classes.textField}          
-          variant="filled"
-          onChange={inputHandler}
-          type="text"
-          name="instruction"
-        />
+      <br></br>
+      <TextField 
+      id="filled-full-width"
```
Comment (carnundotcom) [inline src/components/createRecipe/recipeForm/Step4.jsx:113]: Indent! :)

File: src/components/createRecipe/recipeForm/Step4.jsx [hunk 8]
```diff
@@ -131,19 +136,17 @@ function Step4(props) {
...
-                <AddItem>
-                  <p key={i}>{ing}</p>
-                  <button onClick={ e => {removeInstruction(e, ing, i)} }>X</button>                  
-                </AddItem>
-              ))
-            : null}
-        </div>
+        {instructionsArray.length
+          ? instructionsArray.map((ing, i) => 
+          <AddItem>
```
Comment (carnundotcom) [inline src/components/createRecipe/recipeForm/Step4.jsx:142]: Formatting is also a little awkward here. :)

**Review by carnundotcom — CHANGES_REQUESTED** [review]
Has all this been tested? :)

File: src/components/createRecipe/recipeForm/Step4.jsx (hunk 6 above)
Comment (KiyaniBamba) [reply src/components/createRecipe/recipeForm/Step4.jsx:33]: good idea, but I will probably do it later when the more urgent functionalities will be working!

**Review by carnundotcom — APPROVED** [review]

**Review by ThorbenBender — APPROVED** [review]

**Pull request merged**